components/authz-service/server/v2/project.go

```diff
@@ -158,13 +164,50 @@ func (s *state) UpdateProject(ctx context.Context,
 }
 
 func (s *state) ApplyRulesStart(
-	context.Context, *api.ApplyRulesStartReq) (*api.ApplyRulesStartResp, error) {
-	s.log.Info("apply project rules: START")
-	err := s.projectUpdateManager.Start()
+	ctx context.Context, _ *api.ApplyRulesStartReq) (*api.ApplyRulesStartResp, error) {
+	// NOTE (tc): Only one call to ApplyRulesStart can happen at a time.
+	// This should be good enough to prevent race conditions for single node,
+	// in conjunction with the table locking that happens in store.ApplyStagedRules.
+	// Can still get into a weird state if we panic, but a re-apply will fix things up.
+	// We will be refactoring for multi-node using workflow tooling in the near future.
+	s.applyRuleMux.Lock()
+	defer s.applyRuleMux.Unlock()
+
+	switch s.projectUpdateManager.State() {
+	case config.NotRunningState:
+		break
+	case config.RunningState:
+		return nil, status.Error(codes.FailedPrecondition,
+			"cannot apply rules: apply already in progress")
+	default:
+		return nil, status.Error(codes.Internal,
+			"failed to parse apply state")
```

Comment:
super nit: I'd maybe include the word `rules` or `project rules` or something in addition to `apply` here as `apply` feels vague-- but I don't have a mental model rn of what the error message would be